megamek/src/megamek/client/ui/dialogs/MiniReportDisplayDialog.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2022 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.dialogs;
+
+import megamek.client.ui.Messages;
+import megamek.client.ui.swing.ClientGUI;
+import megamek.client.ui.swing.GUIPreferences;
+import megamek.client.ui.swing.util.UIUtil;
+
+import javax.swing.*;
+import java.awt.event.KeyEvent;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+
+public class MiniReportDisplayDialog extends JDialog {
+    //region Variable Declarations
+    private final ClientGUI clientGUI;
+    private static final GUIPreferences GUIP = GUIPreferences.getInstance();
+    private static final String MSG_TITLE = Messages.getString("MiniReportDisplay.title");
+    //endregion Variable Declarations
+
+    //region Constructors
+    public MiniReportDisplayDialog(final JFrame frame, final ClientGUI clientGUI) {
+        super(frame, MSG_TITLE, false);
+
+        this.setLocation(GUIP.getMiniReportPosX(), GUIP.getMiniReportPosY());
+        this.setSize(GUIP.getMiniReportSizeWidth(), GUIP.getMiniReportSizeHeight());
+
+        UIUtil.updateWindowBounds(this);
+        this.setResizable(true);
+        this.setFocusable(false);
+        this.setFocusableWindowState(false);
+
+        addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowClosing(WindowEvent evt) {
+                GUIP.setMiniReportEnabled(false);
+            }
+        });
+
+        this.clientGUI = clientGUI;
+    }
+    //endregion Constructors
+
+    public void savePref() {
+        GUIP.setMiniReportSizeWidth(getSize().width);
+        GUIP.setMiniReportSizeHeight(getSize().height);
+        GUIP.setMiniReportPosX(getLocation().x);
+        GUIP.setMiniReportPosY(getLocation().y);
+    }
+
+    @Override
+    protected void processWindowEvent(WindowEvent e) {
+        super.processWindowEvent(e);
+        if ((e.getID() == WindowEvent.WINDOW_DEACTIVATED) || (e.getID() == WindowEvent.WINDOW_CLOSING)) {
```

Comment:
Could the savePref() be moved up to the WindowClosing listener with the same result?